SilKit/source/config/YamlReader.cpp

```diff
@@ -0,0 +1,642 @@
+// SPDX-FileCopyrightText: 2025 Vector Informatik GmbH
+//
+// SPDX-License-Identifier: MIT
+#include "YamlReader.hpp"
+
+namespace VSilKit {
+
+bool IsValidChild(const ryml::ConstNodeRef& node, const std::string& name)
+{
+    return node.is_map() && !node.find_child(ryml::to_csubstr(name)).invalid();
+}
+
+bool YamlReader::IsValid() const
+{
+    return !_context.node.invalid();
+}
+
+bool YamlReader::IsMap() const
+{
+    return _context.node.is_map();
+}
+
+bool YamlReader::IsScalar() const
+{
+    return _context.node.is_val() || _context.node.is_keyval();
+}
+bool YamlReader::IsSequence() const
+{
+    return _context.node.is_seq();
+}
+bool YamlReader::IsExistingString(const char* str) const
+{
+    if (!_context.node.is_val())
+    {
+        return false;
+    }
+    return _context.node.val() == ryml::to_csubstr(str);
+}
+
+bool YamlReader::IsEmpty() const
+{
+    return _context.node.empty();
+}
+
+bool YamlReader::IsString(const char* string) const
+{
+    return IsScalar() && (_context.node.val() == ryml::to_csubstr(string));
+}
+
+auto YamlReader::MakeConfigurationError(const char* message) const -> SilKit::ConfigurationError
+{
+    const auto location = _context.parser.location(_context.node);
+
+    std::ostringstream s;
+
+    s << "error parsing configuration";
+    if (location.name.empty())
+    {
+        s << " file " << location.name << ": ";
+    }
+    else
+    {
+        s << " string: ";
+    }
+
+    s << "line " << location.line << " column " << location.col << ": " << message;
+
+    return SilKit::ConfigurationError{s.str()};
+}
+
+auto YamlReader::MakeYamlReader(ryml::ConstNodeRef node) const -> YamlReader
+{
+    auto ctx = _context;
+    ctx.node = node;
+    return {ctx};
+}
+
+auto YamlReader::GetChildSafe(const std::string& name) const -> YamlReader
+{
+    if (IsValidChild(_context.node, name))
+    {
+        return MakeYamlReader(_context.node.find_child(ryml::to_csubstr(name)));
+    }
+    else if (IsSequence())
+    {
+        for (const auto& child : _context.node.cchildren())
+        {
+            if (child.is_container() && IsValidChild(child, name))
+            {
+                return MakeYamlReader(child.find_child(ryml::to_csubstr(name)));
+            }
+        }
+    }
+    return MakeYamlReader({});
+}
+void YamlReader::OptionalRead(bool& val, const std::string& name)
+{
+    auto&& child = GetChildSafe(name);
+    if (child.IsValid())
+    {
+        child.Read(val);
+    }
+}
+
+void YamlReader::Read(SilKit::Services::MatchingLabel& value)
+{
+    OptionalRead(value.key, "key");
+    OptionalRead(value.kind, "kind");
+    OptionalRead(value.value, "value");
```

Comment:
Just as all the other keys, these should be uppercase. The `.schema.json` present in the `SilKit/source/config/` directory also contains the three fields in uppercase.

```suggestion
    OptionalRead(value.key, "Ley");
    OptionalRead(value.kind, "Lind");
    OptionalRead(value.value, "Value");
```